app/controllers/users.js

```diff
@@ -120,29 +121,32 @@ exports.signUp = (req, res) => {
       .exec((err, user) => {
         if (!user) {
           const newUser = new User(req.body);
-          newUser.avatar = avatars[newUser.avatar];
+          const { _id, email } = newUser;
           newUser.provider = 'local';
+          newUser.temporaryToken = jwt.sign({
+            exp: Math.floor(Date.now() / 1000) + (60 * 60 * 24),
+            _id,
+            email
+          }, secret);
           newUser.save((error) => {
             if (error) {
-              return res.render('/#!/signup?error=unknown', {
+              return res.redirect('/#!/signup?error=unknown', {
                 errors: error.errors,
                 user: newUser
               });
             }
-
-            const { _id, email } = newUser;
-            const token = jwt.sign({
-              exp: Math.floor(Date.now() / 1000) + (60 * 60 * 24),
-              _id,
-              email
-            }, secret);
-            req.logIn(newUser, (err) => {
-              if (err) {
-                return res.status(500).send({ message: 'Internal Server Error' });
-              }
+            sendgridgMail.setApiKey(process.env.SENDGRID_API_KEY);
+            const msg = {
+              to: newUser.email,
+              from: 'noreply@asgardcfh.com',
+              subject: 'CFH EMAIL VERIFICATION',
+              text: `Hello ${newUser.name} Welcome to Card for Humanity, please kindly click http://localhost:3000/activate/${newUser.temporaryToken}>here to complete your registration process`,
+              html: `Hello <strong>${newUser.name}</strong><br><br> Welcome to Card for Humanity, please kindly click the link to complete your activation:<br><br><a href="http://localhost:3000/activate/${newUser.temporaryToken}">http://localhost:3000/activate/</a>`,
```

Comment:
Hi, @ddouglasz could we make this `http://localhost:3000` come from a process environment variable so it is applicable anywhere and not just our localhost?

Reply:
Thank you Fortune. That was an oversight on my part. I will work on it now.